huaweicloud/services/cph/resource_huaweicloud_cph_phone_stop.go

```diff
@@ -0,0 +1,135 @@
+)
+
+var PhoneStopNonUpdatableParams = []string{"phone_ids"}
+
+// @API CPH POST /v1/{project_id}/cloud-phone/phones/batch-stop
+// @API CPH GET /v1/{project_id}/cloud-phone/jobs
+func ResourcePhoneStop() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourcePhoneStopCreate,
+		UpdateContext: resourcePhoneStopUpdate,
+		ReadContext:   resourcePhoneStopRead,
+		DeleteContext: resourcePhoneStopDelete,
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		CustomizeDiff: config.FlexibleForceNew(PhoneStopNonUpdatableParams),
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"phone_ids": {
+				Type:        schema.TypeList,
+				Required:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Description: `Specifies the IDs of the CPH phone.`,
+			},
+			"enable_force_new": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringInSlice([]string{"true", "false"}, false),
+				Description:  utils.SchemaDesc("", utils.SchemaDescInput{Internal: true}),
+			},
+		},
+	}
+}
+
+func resourcePhoneStopCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	client, err := cfg.NewServiceClient("cph", region)
+	if err != nil {
+		return diag.Errorf("error creating CPH client: %s", err)
+	}
+
+	// createPhoneStop: create CPH phone stop
+	createPhoneStopHttpUrl := "v1/{project_id}/cloud-phone/phones/batch-stop"
+	createPhoneStopPath := client.Endpoint + createPhoneStopHttpUrl
+	createPhoneStopPath = strings.ReplaceAll(createPhoneStopPath, "{project_id}", client.ProjectID)
+
+	createPhoneStopOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		JSONBody: map[string]interface{}{
+			"phone_ids": d.Get("phone_ids"),
+		},
+	}
+
+	createPhoneStopResp, err := client.Request("POST", createPhoneStopPath, &createPhoneStopOpt)
+	if err != nil {
+		return diag.Errorf("error creating CPH phone stop: %s", err)
+	}
+
+	resp, err := utils.FlattenResponse(createPhoneStopResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	id := utils.PathSearch("request_id", resp, "").(string)
+	if id == "" {
+		return diag.Errorf("Unable to find the request ID from the API response")
+	}
+	d.SetId(id)
+
+	jobs := utils.PathSearch("jobs", resp, make([]interface{}, 0)).([]interface{})
+	for _, v := range jobs {
+		errorCode := utils.PathSearch("error_code", v, "").(string)
+		if errorCode != "" {
+			serverId := utils.PathSearch("phone_id", v, "").(string)
+			errorMsg := utils.PathSearch("error_msg", v, "").(string)
+			log.Printf("[WARN] Failed to stop CPH phone (phone_id: %s), error_code: %s, error_msg: %s", serverId, errorCode, errorMsg)
+		}
+	}
+
+	err = checkJobStatus(ctx, client, d.Id(), d.Timeout(schema.TimeoutCreate))
```

Comment:
the d.Id() is the request_id,  not the job_id

Reply:
fixed，use to get phone detail API status
![image](https://github.com/user-attachments/assets/cbe1cb21-a10f-4c60-8f91-22da57cb5e71)